hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -291,55 +296,81 @@ public List<DatanodeDetails> filterNodesWithSpace(List<DatanodeDetails> nodes,
   }
 
   /**
-   * Returns true if this node has enough space to meet our requirement.
+   * Check if a datanode has sufficient space and return detailed result.
    *
    * @param datanodeDetails DatanodeDetails
-   * @return true if we have enough space.
+   * @param metadataSizeRequired metadata size required
+   * @param dataSizeRequired data size required
+   * @param conf configuration
+   * @return SCMDatanodeSpaceCheckResult with detailed volume information
    */
-  public static boolean hasEnoughSpace(DatanodeDetails datanodeDetails,
-                                       long metadataSizeRequired,
-                                       long dataSizeRequired,
-                                       ConfigurationSource conf) {
+  public static SCMDatanodeSpaceCheckResult checkSpace(DatanodeDetails datanodeDetails,
+                                                       long metadataSizeRequired,
+                                                       long dataSizeRequired,
+                                                       ConfigurationSource conf) {
```

Comment:
nit: Please do not format method signature like this. Whenever visibility / return type / method name / other modifiers are changed, we would have to reindent all parameters.